dialect_common_go1.8.go

```diff
@@ -0,0 +1,36 @@
+// +build go1.8
+
+package gorm
+
+import (
+	"context"
+	"fmt"
+)
+
+func (s commonDialect) HasIndex(tableName string, indexName string) bool {
+	var count int
+	s.db.QueryRowContext(context.Background(), queryHasIndex, s.CurrentDatabase(), tableName, indexName).Scan(&count)
```

Comment:
this applies for all the places where this occurs

Reply:
We don't have access to `*DB.Context()` in dialects codes. We can add context to them with the same approach, but it is a lot of extra work. The changeset is already huge for just adding context to the main structure, because of the backward compatibility with Go<`1.8`.

If it's ok, I will update the pull request and send it by the end of today.

@sarathsp06 @sergiosalvatore

Reply:
agreed @remohammadi. let me know if there's anything I can do to help...